src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/opsBreakdown.tsx

```diff
@@ -79,82 +82,120 @@ class OpsBreakdown extends React.Component<Props> {
             },
           ];
 
-    type AggregateType = {
-      [opname: string]: {
-        totalDuration: number; // num of seconds
-      };
-    };
+    const operationNameIntervals = spans.reduce(
+      (intervals: OperationNameIntervals, span: RawSpanType) => {
+        let startTimestamp = span.start_timestamp;
+        let endTimestamp = span.timestamp;
 
-    let cumulativeDuration = 0;
+        if (endTimestamp < startTimestamp) {
+          // reverse timestamps
+          startTimestamp = span.timestamp;
+          endTimestamp = span.start_timestamp;
+        }
 
-    const aggregateByOp: AggregateType = spans.reduce(
-      (aggregate: AggregateType, span: RawSpanType) => {
-        let op = span.op;
+        // invariant: startTimestamp <= endTimestamp
 
-        const duration = Math.abs(span.timestamp - span.start_timestamp);
-        cumulativeDuration += duration;
+        let operationName = span.op;
 
-        if (typeof op !== 'string') {
+        if (typeof operationName !== 'string') {
           // a span with no operation name is considered an 'unknown' op
-          op = 'unknown';
+          operationName = 'unknown';
         }
-        const opStats = aggregate[op];
 
-        if (!opStats) {
-          aggregate[op] = {
-            totalDuration: duration,
-          };
-          return aggregate;
+        const cover: TimeWindowSpan = [startTimestamp, endTimestamp];
+
+        const operationNameInterval = intervals[operationName] ?? [];
+
+        if (operationNameInterval.length <= 0) {
+          intervals[operationName] = [cover];
```

Comment:
Couldn't this also be a push()? You initialize an empty array on 107 and then build another one here.

Reply:
It doesn't work when `intervals[operationName]` is undefined.

Reply:
Then why build the array on 107? You could use an undefined check.

Reply:
should be addressed in https://github.com/getsentry/sentry/pull/18522/commits/bf7dd0ccb24b44935c0b493ee9d44af137e7d864